cmd/root_test.go

```diff
@@ -29,100 +29,132 @@ import (
+	c.Assert(err, IsNil)
+
+	// find repo when workDir is another git repo under ./hpc-toolkit
+	dir := ".." + "/hpc-toolkit/modules/network/vpc/.terraform/modules/cloud_router/.git"
+	err = os.Chdir(dir)
+	_, got, err = hpcToolkitRepo()
+	c.Assert(got, Equals, repoDir)
+	c.Assert(err, IsNil)
+
+	// Try to find repo when workdir is outside root of repo. Normal execution
+	// returns binary site in. Empty value expected during tests since test
+	// binary is built in a test dir that is not in a git repository
+	err = os.Chdir(filepath.Dir(repoDir))
 	_, got, _ = hpcToolkitRepo()
 	c.Assert(got, Equals, "")
 }
 
-func (m *MySuite) TestIsHpcToolkitRepo(c *C) {
+func (s *MySuite) TestIsHpcToolkitRepo(c *C) {
+	// sub directory of an hpc-toolkit git repository
 	_, callDir, _, _ := runtime.Caller(0)
 	repoDir := filepath.Dir(filepath.Dir(callDir))
 	repo, _ := git.PlainOpen(repoDir)
-
 	got := isHpcToolkitRepo(*repo)
 	c.Assert(got, Equals, true)
 
-	testRepo, _ := git.Init(memory.NewStorage(), fs)
-
-	initTestRepo(*testRepo)
+	// temporary git repo is not a hpc-toolkit repo
+	storer := memory.NewStorage()
+	fs := memfs.New()
+	testRepo, _ := git.Init(storer, fs)
+	initTestRepo(*testRepo, fs)
 	got = isHpcToolkitRepo(*testRepo)
 	c.Assert(got, Equals, false)
-
-}
-
-func initTestRepo(r git.Repository) {
-	// Create new file
-	filePath := "test.txt"
-	testFile, err := fs.Create(filePath)
-	if err != nil {
-		return
-	}
-	testFile.Write([]byte("Test file"))
-	testFile.Close()
-
-	w, err := r.Worktree()
-
-	w.Add(filePath)
-
-	// git commit -m $message
-	w.Commit("Initial commit", &git.CommitOptions{})
 }
 
-func (m *MySuite) TestCheckGitHashMismatch(c *C) {
+func (s *MySuite) TestCheckGitHashMismatch(c *C) {
 	_, callDir, _, _ := runtime.Caller(0)
 	repoDir := filepath.Dir(filepath.Dir(callDir))
+	workDir := filepath.Dir(callDir)
 	repo, _ := git.PlainOpen(repoDir)
 	head, _ := repo.Head()
-	expected := head.Hash().String()
+	hash := head.Hash().String()
+	branch := head.Name().Short()
 
 	// verify current working directory git hash against hash of v1.7.0
+	GitInitialHash = "8fc4768edbef9b3f115a41eaf2a5740d41758cff"
 	GitCommitHash = "f4ed7c1ca864c77ea8cc21bd9286b0c48d28db95"
-	_, got, _ := checkGitHashMismatch()
-	c.Assert(got, Equals, expected)
+	mismatch, b, h, dir := checkGitHashMismatch()
+	c.Assert(mismatch, Equals, true)
+	c.Assert(dir, Equals, workDir)
+	c.Assert(b, Equals, branch)
+	c.Assert(h, Equals, hash)
+
+	// verify current working directory git hash against random initial hash
+	GitInitialHash = "6c50deedff0520887a50b8ad2ca84724ff1adae8"
+	GitCommitHash = "f4ed7c1ca864c77ea8cc21bd9286b0c48d28db95"
+	mismatch, b, h, dir = checkGitHashMismatch()
+	c.Assert(mismatch, Equals, false)
+	c.Assert(b, Equals, "")
+	c.Assert(h, Equals, "")
 
 	// verify current working directory git hash against a wrong random hash
+	GitInitialHash = "8fc4768edbef9b3f115a41eaf2a5740d41758cff"
 	GitCommitHash = "6c50deedff0520887a50b8ad2ca84724ff1adae8"
```

Comment:
nit: It took me a minute to wrap my head around these cases. I think readability could be improved with some intermediate variable naming like:

```
randomGitHash := f4ed7c1ca864c77ea8cc21bd9286b0c48d28db95
initalGitHash := 8fc4768edbef9b3f115a41eaf2a5740d41758cff
oldGitHashFromMain := f4ed7c1ca864c77ea8cc21bd9286b0c48d28db95

// and then the setup would read

GitInitialHash = initalGitHash
GitCommitHash = randomGitHash
```